Casks/vlc-nightly.rb

```diff
@@ -4,9 +4,11 @@
 
   url do
     require "open-uri"
-    base_url = "https://nightlies.videolan.org/build/macosx-intel/last/"
-    file = URI(base_url).open.read.scan(/href="([^"]+.dmg)"/).flatten.first
-    "#{base_url}#{file}"
+    base_url = "https://artifacts.videolan.org/vlc/nightly-macos/"
+    folder = URI(base_url).open.read.scan(/(\d+-\d+)/).flatten.first
```

Comment:
I think we do this for other casks as well, but there’s no point to using `scan` if we’re going to use the first result. `match` does it fine.

```suggestion
    folder = URI(base_url).open.read.match(/(\d+-\d+)/).to_s
```

Reply:
Done.

Reply:
Same thing can be done below.

Reply:
Also done.

Reply:
Or just `.read[/(\d+-\d+)/, 1]`.

Reply:
> Or just `.read[/(\d+-\d+)/, 1]`.

That is better -- makes it clear that it is the _first_ match in the list. Thanks @reitermarkus.

Reply:
> Or just `.read[/(\d+-\d+)/, 1]`

What’s the `, 1` for?

Reply:
> > Or just `.read[/(\d+-\d+)/, 1]`
> 
> What’s the `, 1` for?

First match.

Reply:
> First match.

It’s always the first match, there’s only one capture being made. For the `1` to be meaningful, there would need to be more than one set of parenthesis. Which we can also remove, because we don’t need a capture group in this case: `URI(base_url).read[/\d+-\d+/]`

Reply:
Good point -- updated.